pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -117,11 +117,23 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 
 	tlsConn := tls.Client(rwc, s.tlsConfig)
 
+	// Return if we were configured to skip validation
+	if s.tlsConfig != nil && s.tlsConfig.InsecureSkipVerify {
+		return tlsConn, nil
+	}
+
 	// need to manually call Handshake() so we can call VerifyHostname() below
 	if err := tlsConn.Handshake(); err != nil {
 		return nil, err
 	}
 
+	state := tlsConn.ConnectionState()
+	for _, peerCert := range state.PeerCertificates {
+		if err := peerCert.VerifyHostname(host); err != nil {
```

Comment:
Ok, I think this is probably not needed here, but will be needed in cases where we call `VerifyHostname` before/without doing a handshake. Let me check if there are places like that and add tests for all these scenarios to make sure we confirm the theory.